### DISCUSSION TITLE
Corner Radius on TextBlock?

### DISCUSSION THREAD

**Discussion by Anequit** [AvaloniaUI/Avalonia#6358] [Q&A] [answered]
Is this possible? I've tried doing it through a style and it's not working.

**Comment by maxkatz6** [comment, marked as answer]
It is possible with a style.
I assume you are working with Fluent theme, which is enabled by default in avalonia templates. For Fluent theme you need this style:
```
  <Style Selector="TextBox /template/ Border#PART_BorderElement">
    <Setter Property="CornerRadius" Value="{DynamicResource ControlCornerRadius}" />
  </Style>
```

For old-style default theme it should be this one:
```
  <Style Selector="TextBox /template/ Border#border">
    <Setter Property="CornerRadius" Value="{DynamicResource ControlCornerRadius}" />
  </Style>
```

**Reply by maxkatz6** [reply]
Or you can wait for https://github.com/AvaloniaUI/Avalonia/pull/6347

**Reply by Anequit** [reply]
it works with textbox alright, but with textblock it doesn't

**Reply by maxkatz6** [reply]
Oh, I misunderstood the question. 
TextBlock is just a text block, it doesn't have any logic for background or border. 
You need to add a Border control to be a parent of text block.

**Reply by Anequit** [reply]
That sounds like more work than just turning a textbox into a textblock 😆

**Reply by maxkatz6** [reply]
No... It's just a text block inside of the border:
```xaml
<Border Background="Green" CornerRadius="4" >
    <TextBlock />
</Border>
```

**Reply by Anequit** [reply]
Well then... I guess that is easier than turning a textbox into a textblock...
